lib/javascript.py

```diff
@@ -0,0 +1,401 @@
+
+    # this block is for require (as in: import) expressions
+    if target['init']['type'] == 'CallExpression' \
+       and target['init']['callee'].get('name') == 'require':
+        import_src_str = target['init']['arguments'][0]['value']
+        if 'name' in target['id']:
+            imported_name = target['id']['name']
+            points_to_str = djoin(import_src_str, imported_name)
+            return [Variable(imported_name, points_to_str, lineno(element))]
+        ret = []
+        for prop in target['id'].get('properties', []):
+            imported_name = prop['key']['name']
+            points_to_str = djoin(import_src_str, imported_name)
+            ret.append(Variable(imported_name, points_to_str, lineno(element)))
+        return ret
+
+    # For the other type of import expressions
+    if target['init']['type'] == 'ImportExpression':
+        import_src_str = target['init']['source']['raw']
+        imported_name = target['id']['name']
+        points_to_str = djoin(import_src_str, imported_name)
+        return [Variable(imported_name, points_to_str, lineno(element))]
+
+    if target['init']['type'] == 'CallExpression':
+        if 'name' not in target['id']:
+            return []
+        call = get_call_from_func_element(target['init'])
+        return [Variable(target['id']['name'], call, lineno(element))]
+
+    if target['init']['type'] == 'ThisExpression':
+        assert set(target['init'].keys()) == {'start', 'end', 'loc', 'type'}
+        return []
+    return []
+
+
+def make_local_variables(tree, parent):
+    """
+    Given an ast of all the lines in a function, generate a list of
+    variables in that function. Variables are tokens and what they link to.
+    In this case, what it links to is just a string. However, that is resolved
+    later.
+
+    Also return variables for the outer scope parent
+
+    :param tree list|dict:
+    :param parent Group:
+    :rtype: list[Variable]
+    """
+    if not tree:
+        return []
+
+    variables = []
+    for element in walk(tree):
+        if element['type'] == 'VariableDeclaration':
+            variables += process_assign(element)
+
+    # Make a 'this' variable for use anywhere we need it that points to the class
+    if isinstance(parent, Group) and parent.group_type == 'CLASS':
+        variables.append(Variable('this', parent, lineno(tree)))
+
+    variables = list(filter(None, variables))
+    return variables
+
+
+def children(tree):
+    """
+    The acorn AST is tricky. This returns all the children of an element
+    :param ast tree:
+    :rtype: list[ast]
+    """
+    assert type(tree) == dict
+    ret = []
+    for k, v in tree.items():
+        if type(v) == dict and v.get('type'):
+            ret.append(v)
+        if type(v) == list:
+            ret += v
+    return ret
+
+
+def get_acorn_version():
+    """
+    Get the version of installed acorn
+    :rtype: str
+    """
+    return subprocess.check_output(['npm', '-v', 'acorn'])
```

Comment:
`npm -v acorn` returns the version of **npm** itself. You can find the version of  the **installed** package with `npm list` or `node -p` as bellow;
```console
$ npm install acorn@8.1.0

$ npm list --depth=0 acorn
/home/foo/code2flow
└── acorn@8.1.0

$ node -p "require('acorn/package.json').version"
8.1.0
```

Reply:
That was a big oversight on my part. Thank you! Fixed.